std/bitmanip.d

```diff
@@ -4005,3 +4005,444 @@ auto bitsSet(T)(T value) @nogc pure nothrow
+    }
+    else
+    {
+        alias minUnsignedIntegral = T;
+    }
+}
+
+/** Tests if the bit `idx` is set in integer `bitfield`.
+If `idx >= T.sizeof * 8` the bahaviour is undefined.
+
+Params:
+   bitfield = the integer containing bit to test
+   idx = the index of the bit to test
+
+Returns: `true` if the bit is set, `false` otherwise.
+*/
+bool testBit(T)(const T bitfield, const ulong idx) if (isIntegral!T)
+{
+    return (cast(ulong)(bitfield) & (1UL << idx)) > 0UL;
+}
+
+///
+unittest
+{
+    int a = 0b0000_0000_0010_0001;
+    assert(testBit(a, 0));
+    assert(!testBit(a, 1));
+    assert(testBit(a, 5));
+    assert(!testBit(a, 6));
+}
+
+/** Tests if any bit is set in the integer `bitfield`.
+
+Params:
+   bitfield = the integer to test
+
+Returns: `true` if any bit is set, `false otherwise.
+*/
+bool testAnyBit(T)(const T bitfield) if (isIntegral!T)
+{
+    return bitfield != T.init;
+}
+
+///
+unittest
+{
+    int a = 0b0000_0000_0010_0001;
+    int b = 0b0000_0000_0000_0000;
+    assert(testAnyBit(a));
+    assert(!testAnyBit(b));
+}
+
+/** Tests if no bit is set in the integer `bitfield`.
+
+Params:
+   bitfield = the integer to test
+
+Returns: `true` if no bits are set, `false otherwise.
+*/
+bool testNoBit(T)(const T bitfield) if (isIntegral!T)
+{
+    return bitfield == T.init;
+}
+
+///
+unittest
+{
+    int a = 0b0000_0000_0010_0001;
+    int b = 0b0000_0000_0000_0000;
+    assert(testAnyBit(a));
+    assert(!testAnyBit(b));
+}
+
+/** Tests if all bits are set in the integer `bitfield`.
+
+Params:
+   bitfield = the integer to test
+
+Returns: `true` if all bits are set, `false otherwise.
+*/
+bool testAllBit(T)(const T bitfield) if (isIntegral!T)
+{
+    return cast(minUnsignedIntegral!T)(bitfield) == minUnsignedIntegral!T.max;
+}
+
+///
+unittest
+{
+    int a = 0b1111_1111_0010_1111;
+    int b = 0b1111_1111_1111_1111;
+    assert(!testAllBit(a));
+    assert(testAnyBit(b));
+}
+
+unittest
+{
+    import std.meta : AliasSeq;
+
+    foreach (T; AliasSeq!(ubyte, ushort, uint, ulong))
+    {
+        T min = T.min;
+        T max = T.max;
+
+        for (size_t i = 0; i < T.sizeof * 8; ++i)
+        {
+            assert(!testBit(min, i));
+            assert(testBit(max, i));
+        }
+    }
+
+    foreach (T; AliasSeq!(byte, short, int, long))
+    {
+        T min = T.min;
+        T max = T.max;
+
+        for (size_t i = 0; i < (T.sizeof * 8) - 1; ++i)
+        {
+            assert(!testBit(min, i));
+            assert(testBit(max, i));
+        }
+        assert(testBit(min, T.sizeof * 8 - 1));
+        assert(!testBit(max, T.sizeof * 8 - 1));
+    }
+}
+
+/** Creates a copy of the integer `bitfield` and sets the bit `idx` in the
+returned copy.  If `idx >= T.sizeof * 8` the bahaviour is undefined.
+
+Params:
+   bitfield = the integer to modify
+   idx = the index of the bit to set
+
+Returns: a copy of `bitfield` with bit `idx` set
+*/
+T setBit(T)(const T bitfield, const ulong idx) if (isIntegral!T)
+{
```

Comment:
> The difference should disappear with inlining anyway. (Still not a fan of having them in the first place. ^^)

@klickverbot I tried both [separate files] (https://github.com/wilzbach/d-benchmarks) and [one single file](https://gist.github.com/wilzbach/c8dd833fd6f2df43604b9c6b0cd0dff4) and the CTFE difference doesn't disappear.

> Of course it is faster, I never doubted that. But how often do you think people will have a function where they know at CT what bit they want to set, but the value they want to set it in is a RT value.  And even then people can use this and CTFE to build a bitmask and than just | or &

I thought that's the common case as the programmer usually knows his flags at CT, but wants to test/set them at RT.

> Additionally, someone already said this PR was overengineering, what chance of getting this in do you think will be there if I add partial CTFE versions of the functions $\lim_{x = 0} $

Understood :/